generators/entity-client/templates/angular/src/main/webapp/app/entities/_entity.model.ts

```diff
@@ -70,8 +70,29 @@ relationships.forEach(relationship => {
             }
             fieldName = relationship.relationshipFieldName;
         } else {
-            fieldType = tsKeyType;
-            fieldName = `${relationship.relationshipFieldName}Id`;
+            const otherEntityRelationshipName = relationship.otherEntityRelationshipName;
+            const relationshipFieldName = relationship.relationshipFieldName;
+            const relationshipFieldNamePlural = relationship.relationshipFieldNamePlural;
+            const relationshipType = relationship.relationshipType;
+            const otherEntityNameCapitalized = relationship.otherEntityNameCapitalized;
+            const otherEntityFieldCapitalized = relationship.otherEntityFieldCapitalized;
+            const ownerSide = relationship.ownerSide;
+
+            if (relationshipType === 'many-to-many' && ownerSide === true) {
+                fieldType = otherEntityNameCapitalized + '[]';
+                fieldName = relationshipFieldNamePlural;
+            } else if (relationshipType === 'many-to-one' || (relationshipType === 'one-to-one' && ownerSide === true)) {
+                if (otherEntityFieldCapitalized !='Id' && otherEntityFieldCapitalized !== '') {
+                    fieldType = 'string';
+                    fieldName = relationshipFieldName + otherEntityFieldCapitalized;
```

Comment:
same here
`${relationshipFieldName}${otherEntityFieldCapitalized}` and `${fieldName}?: ${fieldType}`

Reply:
Agree